qhub/initialize.py

```diff
@@ -508,3 +497,62 @@ def auth0_auto_provision(config):
     config["security"]["authentication"]["config"]["auth0_subdomain"] = auth0_config[
         "auth0_subdomain"
     ]
+
+
+def _set_kubernetes_version(
+    config, kubernetes_version, cloud_provider, grab_latest_version=True
+):
+    cloud_provider_dict = {
+        "aws": {
+            "full_name": "amazon_web_services",
+            "k8s_version_checker_func": amazon_web_services.kubernetes_versions,
+        },
+        "azure": {
+            "full_name": "azure",
+            "k8s_version_checker_func": azure_cloud.kubernetes_versions,
+        },
+        "do": {
+            "full_name": "digital_ocean",
+            "k8s_version_checker_func": digital_ocean.kubernetes_versions,
+        },
+        "gcp": {
+            "full_name": "google_cloud_platform",
+            "k8s_version_checker_func": google_cloud.kubernetes_versions,
+        },
+    }
+    cloud_full_name = cloud_provider_dict[cloud_provider]["full_name"]
+    func = cloud_provider_dict[cloud_provider]["k8s_version_checker_func"]
+    cloud_config = config[cloud_full_name]
+
+    def _raise_value_error(cloud_provider, k8s_versions):
+        raise ValueError(
+            f"\nInvalid `kubernetes-version` provided: {kubernetes_version}.\nPlease select from one of the following {cloud_provider.upper()} supported Kubernetes versions: {k8s_versions} or omit flag to use latest Kubernetes version available."
+        )
+
+    def _check_and_set_kubernetes_version(
+        kubernetes_version=kubernetes_version,
+        cloud_provider=cloud_provider,
+        cloud_config=cloud_config,
+        func=func,
+    ):
+        region = cloud_config["region"]
+
+        # to avoid using cloud provider SDK
+        # set QHUB_K8S_VERSION environment variable
+        if not QHUB_K8S_VERSION:
+            k8s_versions = func(region)
```

Comment:
What`s happening here if the env variable isn't available?

Reply:
If this env variable is not set, then the `else` branch is executed which performs the actual kubernetes version check. This env variable was included as way to bypass these checkers while developing.